### PR TITLE
#2415 Added PetalBot

### LINKED ISSUE #2415: Add "PetalBot" to the list of crawlers
We get a lot of "search crawler" requests with the following header - 

Mozilla/5.0 (Linux; Android 7.0;) AppleWebKit/537.36 (KHTML, like Gecko) Mobile Safari/537.36 (compatible; PetalBot;+https://aspiegel.com/petalbot)

Please add it to the list

I presume it should be something like: Mozilla/5.0 (compatible;PetalBot;+https://aspiegel.com/petalbot)

### PR DESCRIPTION
**Pull request by skoshelev**
Hi. I added the PetalBot to the crawlers list, could you please check my PR and give me feedback

### CODE REVIEW

File: tests/issues/issue-2415.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+        'properties' => [
+            'Browser' => 'PetalBot',
+            'Browser_Type' => 'Bot/Crawler'
+        ],
+        'lite' => false,
+        'standard' => true,
+        'full' => true,
+    ]
+];
+?>
```
Comment (mimmi20) [inline tests/issues/issue-2415.php]: Please remove the closing PHP-Tags, It is not required for PHP-Files. Please add a single empty line at the end of the file.

File: resources/user-agents/crawlers/search-engines/petalbot.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      "browser": "petalbot",
+      "properties": {
+        "Parent": "DefaultProperties",
+        "Comment": "PetalBot"
+      },
+      "children": [
+        {
+          "match": "Mozilla\/5.0 (compatible; PetalBot;+https:\/\/aspiegel.com\/petalbot)",
+          "properties": {
+			"Comment": "PetalBot (PC)"
```
Comment (mimmi20) [inline resources/user-agents/crawlers/search-engines/petalbot.json]: Please fix the identification. Please use spaces only.

**Review by asgrim — APPROVED** [review]
@mimmi20 think the CS issue can be ignored here? Probably needs main branch looking at separately.

File: resources/user-agents/crawlers/search-engines/petalbot.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+          }
+        },
+        {
+          "match": "Mozilla\/5.0 (Linux; Android 7.0;) AppleWebKit\/537.36 (KHTML, like Gecko) Mobile Safari\/537.36 (compatible; PetalBot;+https:\/\/aspiegel.com\/petalbot)",
```
Comment (mimmi20) [inline resources/user-agents/crawlers/search-engines/petalbot.json]: Please make this rule more generic and not escape the Slashes.

Something like "Mozilla/5.0 (Linux; Android 7.*) AppleWebKit/* (KHTML, like Gecko) *Safari/* (compatible; PetalBot*"

We also have placeholder for platforms. Please look into the rules for the Chrome/Android, how to use.

File: resources/user-agents/crawlers/search-engines/petalbot.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    {
+      "userAgent": "PetalBot",
+      "browser": "petalbot",
+      "properties": {
+        "Parent": "DefaultProperties",
+        "Comment": "PetalBot"
+      },
+      "children": [
+        {
+          "match": "Mozilla\/5.0 (compatible; PetalBot;+https:\/\/aspiegel.com\/petalbot)",
```
Comment (mimmi20) [inline resources/user-agents/crawlers/search-engines/petalbot.json]: Please make this rule more generic and not escape the Slashes.

Something like "Mozilla/5.0 (compatible; PetalBot*"

Comment (mimmi20) [inline resources/user-agents/crawlers/search-engines/petalbot.json]: Please add a test for this useragent too.

**Review by mimmi20 — CHANGES_REQUESTED** [review]

**Comment by skoshelev** [comment]
Hi @mimmi20. Thank you for your review, I made suggested changes

**Review by mimmi20 — APPROVED** [review]

**Pull request merged**